### PR TITLE
Ilsgateway: Loss adjustments ils

### PR DESCRIPTION
**Pull request by kkrampa**

### CODE REVIEW

File: custom/ilsgateway/tasks.py [hunk 1]
```diff
@@ -150,13 +152,38 @@ def get_delivery_group_reports(domain, endpoint, facilities, checkpoint, date, l
         offset = 0
 
 
-# Temporary for staging
-@task(queue='background_queue', ignore_result=True)
-def ils_clear_stock_data_task():
-    StockTransaction.objects.filter(report__domain='ilsgateway-test-1').delete()
-    StockReport.objects.filter(domain='ilsgateway-test-1').delete()
-    products = Product.ids_by_domain('ilsgateway-test-1')
+@task(queue='background_queue')
```
Comment (czue) [inline custom/ilsgateway/tasks.py]: did you intentionally remove `ignore_result`? it doesn't look like it's needed

File: custom/ilsgateway/tasks.py [hunk 2]
```diff
@@ -150,13 +152,38 @@ def get_delivery_group_reports(domain, endpoint, facilities, checkpoint, date, l
...
+        StockTransaction.objects.filter(case_id=supply_point).delete()
+    StockReport.objects.filter(domain=domain).delete()
+    StockDataCheckpoint.objects.filter(domain=domain).delete()
+
+
+@task(queue='background_queue')
+def clear_report_data(domain):
+    locations_ids = SQLLocation.objects.filter(domain=domain).values_list('location_id', flat=True)
+    for location_id in locations_ids:
+        GroupSummary.objects.filter(org_summary__supply_point=location_id).delete()
```
Comment (czue) [inline custom/ilsgateway/tasks.py]: I think these can be slightly optimized by using `org_summary__supply_point__in=location_ids` and not bothering with the loop. does that not work?

Comment (czue) [inline custom/ilsgateway/tasks.py]: (applies to most of the loops below)

File: custom/ilsgateway/tasks.py [hunk 3]
```diff
@@ -150,13 +152,38 @@ def get_delivery_group_reports(domain, endpoint, facilities, checkpoint, date, l
...
+    DeliveryGroupReport.objects.filter(supply_point__in=locations.values_list('location_id', flat=True)).delete()
+    products = Product.ids_by_domain(domain)
     StockState.objects.filter(product_id__in=products).delete()
+    for supply_point in locations.exclude(supply_point_id__is_null=True).values_list('supply_point_id', flat=True):
+        StockTransaction.objects.filter(case_id=supply_point).delete()
+    StockReport.objects.filter(domain=domain).delete()
+    StockDataCheckpoint.objects.filter(domain=domain).delete()
+
+
+@task(queue='background_queue')
```
Comment (czue) [inline custom/ilsgateway/tasks.py]: i think this can also have `ignore_result=True`

File: corehq/ex-submodules/casexml/apps/stock/consumption.py [hunk 4]
```diff
@@ -138,7 +138,8 @@ def get_transactions(case_id, product_id, section_id, window_start, window_end):
     SimpleTransaction = collections.namedtuple('SimpleTransaction', ['action', 'value', 'received_on'])
 
     def _to_consumption_tx(txn):
-        if txn.type in (const.TRANSACTION_TYPE_STOCKONHAND, const.TRANSACTION_TYPE_STOCKOUT):
+        if txn.type in (const.TRANSACTION_TYPE_STOCKONHAND, const.TRANSACTION_TYPE_STOCKOUT,
+                        const.TRANSACTION_TYPE_LA):
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/stock/consumption.py]: can you clarify how these are treated in the consumption calculation? I would have thought that they would be more similar to `consumption` than `stockonhand`

Comment (kkrampa) [inline corehq/ex-submodules/casexml/apps/stock/consumption.py]: In ILS they are completely ignored in consumption calculation(https://github.com/dimagi/rapidsms-logistics/blob/master/logistics/consumption.py#L67) so I leave the same behavior in commtrack. I have to add this here due to assertion below (https://github.com/dimagi/commcare-hq/pull/6403/files#diff-316b40eb9efd4d18615faac9aae10ed0L144)

**Comment by kkrampa** [comment]
@czue Updated

File: corehq/ex-submodules/casexml/apps/stock/consumption.py (hunk 4 above)
Comment (czue) [inline corehq/ex-submodules/casexml/apps/stock/consumption.py]: @rowenaluk do you want to weigh in on this? should we ask the tz team whether we want to change this?

Comment (czue) [inline corehq/ex-submodules/casexml/apps/stock/consumption.py]: (merging for now but making a ticket to follow up on it)

**Pull request merged**